src/lib/validate-declarative-schema.js

```diff
@@ -16,7 +16,21 @@ const TargetSchema = joi.array().items(
     appliesTo: joi.string().valid('contacts', 'reports').required(),
     appliesToType: joi.array().items(joi.string()).optional(),
     appliesIf: joi.function().optional(),
-    passesIf: joi.alternatives().conditional('type', { is: 'percent', then: joi.function().required(), otherwise: joi.function().forbidden() }),
+    groupBy: joi.function().optional(),
+    passGroupWithCount: joi.alternatives().conditional('groupBy', {
+      is: joi.exist(),
+      then: joi.number().required(),
+      otherwise: joi.forbidden(),
```

Comment:
If we assume that grouping by the contact is the default, then it might be useful to allow `passGroupWithCount` even when `groupBy` is not specified. The usual way to achieve this would be to iterate over all reports and only emit if you find more than _x_ that match but it might be cleaner to emit all that match and use `passGroupWithCount` to only pass if _x_ are found.

That would mean the name has to change, say `passWithCount`.

What do you think?

Reply:
I think this would break existing targets if we `groupBy` the contact by default. Targets with `appliesTo: 'report'` (which is common), would change behavior quite a bit. So there is a bit to think through there...

In general, target emissions are much smaller than they were -- but they are still held in memory in the webapp. We should err toward patterns which use the minimum number of emissions to count things. The `groupBy` strategy you propose may be intermittently convenient, but I think we should encourage existing patterns.